nodebuilder/settings.go

```diff
@@ -69,7 +69,14 @@ func WithPyroscope(endpoint string, nodeType node.Type) fx.Option {
 					pyroscope.ProfileInuseSpace,
 					pyroscope.ProfileGoroutines,
 				},
-			})
+			}
+
+			if basicAuthUser != "" && basicAuthPassword != "" {
```

Comment:
Nit: not sure do we need this? By default `config.BasicAuthUser` and `config.BasicAuthPassword` are already `""`. Looks like we can do this without `if`.

Reply:
Makes sense, changed that 👍